components/tikv_util/src/metrics/mod.rs

```diff
@@ -115,6 +186,100 @@ pub fn convert_record_pairs(m: HashMap<String, u64>) -> RecordPairVec {
         .collect()
 }
 
+/// MetricsCompactLevel defines the level of compact metrics sample data, a higher level of
+/// means smaller data size and more information loss.
+#[derive(Serialize_repr, Deserialize_repr, Clone, Copy, PartialEq, Eq, Debug)]
+#[repr(u32)]
+pub enum MetricsCompactLevel {
+    // return full original data.
+    NoCompact = 0,
+    // this level try to compact sample without infromation loss.
+    // currently only filter counter with 0 value and histogram with 0 samples.
+    LoseLess = 1,
+    // this level try to reduce the data size as much as possible.
+    // this level also compact histogram vector type by remove histograms which sample count is
+    // smaller than 1% of the max sample count.
+    Aggressive = 2,
+}
+
+impl From<u32> for MetricsCompactLevel {
+    fn from(v: u32) -> Self {
+        match v {
+            0 => Self::NoCompact,
+            1 => Self::LoseLess,
+            2 => Self::Aggressive,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl From<MetricsCompactLevel> for ConfigValue {
+    fn from(l: MetricsCompactLevel) -> ConfigValue {
+        ConfigValue::U32(l as u32)
+    }
+}
+
+impl From<ConfigValue> for MetricsCompactLevel {
+    fn from(c: ConfigValue) -> MetricsCompactLevel {
+        if let ConfigValue::U32(v) = c {
+            v.into()
+        } else {
+            panic!("expect: ConfigValue::U32, got: {:?}", c);
+        }
+    }
+}
+
+impl From<&ConfigValue> for MetricsCompactLevel {
+    fn from(c: &ConfigValue) -> Self {
+        c.clone().into()
+    }
+}
+
+// MetricsLevel controls how much metrics can be returned. A higher level means less data.
+#[derive(Serialize_repr, Deserialize_repr, Clone, Copy, PartialEq, Eq, Debug)]
+#[repr(u32)]
+pub enum MetricsLevel {
```

Comment:
I think we would better not using both frequency and priority in the levels, which might make people confused. 
One possible proposal:
pub enum MetricLevel {
    None,     // return nothing (reserved)
    HighPriority,   // return high priority metrics
    MiddlePriority,   //return middle and high priority metrics (reserved)
    LowPriority,  // return low, middle and high priority metrics (reserved)
    ALL  // return all  metrics
}